`staticTokenAuthenticationShouldCreateWorkingConnectionBetweenPublisherAndSubscriber` (`AuthenticationTests`) failing with "Expectation 'subscriber received a location update' unfulfilled."

Seems to pass on the majority (over 50%) of test runs, therefore looks like it's flakey.

Failing from our Android emulation workflow. These links will only survive as long as GitHub keeps the test run logs, so I'll also quote from the output...

[At API Level 29](https://github.com/ably/ably-asset-tracking-android/actions/runs/3684425848/jobs/6235398967#step:4:1015) (https://github.com/ably/ably-asset-tracking-android/pull/841):

```
com.ably.tracking.tests.AuthenticationTests > staticTokenAuthenticationShouldCreateWorkingConnectionBetweenPublisherAndSubscriber[test(AVD) - 10] FAILED 
	java.lang.AssertionError: Expectation 'subscriber received a location update' unfulfilled.
	at com.ably.tracking.test.android.common.Expectation.assertFulfilled(Expectations.kt:55)
Tests on test(AVD) - 10 failed: There was 1 failure(s).

> Task :integration-testing-app:connectedDebugAndroidTest FAILED
```

[At API Level 24](https://github.com/ably/ably-asset-tracking-android/actions/runs/3685077251/jobs/6235588965#step:4:956) (https://github.com/ably/ably-asset-tracking-android/pull/842):

```
com.ably.tracking.tests.AuthenticationTests > staticTokenAuthenticationShouldCreateWorkingConnectionBetweenPublisherAndSubscriber[test(AVD) - 7.0] FAILED 
	java.lang.AssertionError: Expectation 'subscriber received a location update' unfulfilled.
	at com.ably.tracking.test.android.common.Expectation.assertFulfilled(Expectations.kt:55)
```

[At API Level 29](https://github.com/ably/ably-asset-tracking-android/actions/runs/3685230365/jobs/6235939378#step:4:1090) (https://github.com/ably/ably-asset-tracking-android/pull/843):

```
com.ably.tracking.tests.AuthenticationTests > staticTokenAuthenticationShouldCreateWorkingConnectionBetweenPublisherAndSubscriber[test(AVD) - 10] FAILED 
	java.lang.AssertionError: Expectation 'subscriber received a location update' unfulfilled.
	at com.ably.tracking.test.android.common.Expectation.assertFulfilled(Expectations.kt:55)
Tests on test(AVD) - 10 failed: There was 1 failure(s).
```

[At API Level 24](https://github.com/ably/ably-asset-tracking-android/pull/855) (https://github.com/ably/ably-asset-tracking-android/pull/855):

```
com.ably.tracking.tests.AuthenticationTests > staticTokenAuthenticationShouldCreateWorkingConnectionBetweenPublisherAndSubscriber[test(AVD) - 7.0] FAILED 
	java.lang.AssertionError: Expectation 'subscriber received a location update' unfulfilled.
	at com.ably.tracking.test.android.common.Expectation.assertFulfilled(Expectations.kt:55)
```

[At API Level 29](https://github.com/ably/ably-asset-tracking-android/actions/runs/3730953632/jobs/6328603478#step:4:924) (https://github.com/ably/ably-asset-tracking-android/pull/855):

```
com.ably.tracking.tests.AuthenticationTests > staticTokenAuthenticationShouldCreateWorkingConnectionBetweenPublisherAndSubscriber[test(AVD) - 10] FAILED 
	java.lang.AssertionError: Expectation 'subscriber received a location update' unfulfilled.
	at com.ably.tracking.test.android.common.Expectation.assertFulfilled(Expectations.kt:55)
```

--- a/core-sdk/src/main/java/com/ably/tracking/connection/ConnectionConfigurationModels.kt
+++ b/core-sdk/src/main/java/com/ably/tracking/connection/ConnectionConfigurationModels.kt
@@ -143,7 +143,7 @@ private class JwtAuthentication(clientId: String?, configuration: JwtConfigurati
 
 interface TokenParams {
     /**
-     * Time to live of the token.
+     * Time to live of the token, in milliseconds.
      * If set to 0 then the default value will be used.
      */
     val ttl: Long

--- a/integration-testing-app/src/androidTest/java/com/ably/tracking/tests/AuthenticationTests.kt
+++ b/integration-testing-app/src/androidTest/java/com/ably/tracking/tests/AuthenticationTests.kt
@@ -102,7 +102,8 @@ class AuthenticationTests {
         val keyParts = ABLY_API_KEY.split(":")
         val keyName = keyParts[0]
         val keySecret = keyParts[1]
-        val tokenValidityInSeconds = if (tokenParams != null && tokenParams.ttl > 0L) tokenParams.ttl else 3600L
+        val tokenValidityInSeconds =
+            if (tokenParams != null && tokenParams.ttl > 0L) (tokenParams.ttl / 1000L) else 3600L
         val currentTimestampInSeconds = (Date().time / 1000L)
         val tokenExpirationTimestampInSeconds = currentTimestampInSeconds + tokenValidityInSeconds
         val tokenCapability =

--- a/integration-testing-app/src/androidTest/java/com/ably/tracking/tests/AuthenticationTests.kt
+++ b/integration-testing-app/src/androidTest/java/com/ably/tracking/tests/AuthenticationTests.kt
@@ -69,7 +69,7 @@ class AuthenticationTests {
     private fun createTokenRequest(ably: AblyRealtime, requestParameters: TokenParams? = null): TokenRequest {
         val ablyTokenRequest = ably.auth.createTokenRequest(
             Auth.TokenParams().apply {
-                ttl = requestParameters?.ttl ?: 3600L
+                ttl = requestParameters?.ttl ?: (60L * 60L * 1000L) // default to one hour
                 capability = requestParameters?.capability ?: "{\"*\":[\"*\"]}"
                 clientId = CLIENT_ID
                 timestamp = requestParameters?.timestamp ?: Date().time

